api/v1alpha1/job_types.go

```diff
@@ -0,0 +1,13 @@
+package v1alpha1
+
+// Type defines what job type this is.
+type Type string
```

Comment:
I think this is highly confusing with the built-in Go's "type"
```suggestion
// JobType defines what kind a certain job is.
type JobType string
```